src/Agent/NewRelic/Agent/Core/TransactionTraces/SlowestTransactionCollector.cs

```diff
@@ -6,36 +6,48 @@
 using NewRelic.Agent.Core.Configuration;
 using NewRelic.Agent.Core.Transformers.TransactionTransformer;
 using System.Linq;
+using System.Collections.Concurrent;
+using System.Threading;
 
 namespace NewRelic.Agent.Core.TransactionTraces
 {
     public class SlowestTransactionCollector : ITransactionCollector, IDisposable
     {
-        private volatile TransactionTraceWireModelComponents _slowTransaction;
+        private volatile ConcurrentBag<TransactionTraceWireModelComponents> _slowTransactions = new ConcurrentBag<TransactionTraceWireModelComponents>();
 
         protected ConfigurationSubscriber ConfigurationSubscription = new ConfigurationSubscriber();
 
         public void Collect(TransactionTraceWireModelComponents transactionTraceWireModelComponents)
         {
-            if (transactionTraceWireModelComponents.Duration <= ConfigurationSubscription.Configuration.TransactionTraceThreshold)
+            if(transactionTraceWireModelComponents == null)
```

Comment:
```suggestion
            if (transactionTraceWireModelComponents == null)
```
Just for consistency